### PR TITLE
resources delete command

### PR DESCRIPTION
**Pull request by Akyiaa**
Story: https://github.com/galasa-dev/projectmanagement/issues/1651

Adding `galasactl resources delete -f my.file.yaml`

Blocked awaiting changes on the server. 

PLEASE DO NOT MERGE YET !

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

**Comment by Akyiaa** [comment]
Error handling and collation of errors list improved and merged: https://github.com/galasa-dev/cli/pull/189

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: test-scripts/resources-tests.sh [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+
+    output_file="$ORIGINAL_DIR/temp/resources-apply-output.txt"
+    $cmd | tee $output_file
+    if [[ "${rc}" != "0" ]]; then 
+        error "Failed to get property with name used, get command failed."
+        exit 1
+    fi
+
+    # Check that the previous properties set updated the property value
+    cat $output_file | grep "$prop_name_to_update\s+$prop_value_to_update" -q -E
```
Comment (techcobweb) [inline test-scripts/resources-tests.sh:295]: why do this twice ?

File: test-scripts/resources-tests.sh [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+    cmd="$ORIGINAL_DIR/bin/${binary} properties get --namespace ecosystemtest \
+    --name $prop_name \
+    --bootstrap $bootstrap \
+    --log -"
+
+    info "Command is: $cmd"
+
+    output_file="$ORIGINAL_DIR/temp/resources-delete-output.txt"
+    $cmd | tee $output_file
+    if [[ "${rc}" != "0" ]]; then
```
Comment (techcobweb) [inline test-scripts/resources-tests.sh:368]: you don't set rc before you test it ?

File: test-scripts/resources-tests.sh [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+    cmd="$ORIGINAL_DIR/bin/${binary} properties get --namespace ecosystemtest \
+    --name $prop_name \
+    --bootstrap $bootstrap \
+    --log -"
+
+    info "Command is: $cmd"
+
+    output_file="$ORIGINAL_DIR/temp/resources-create-output.txt"
+    $cmd | tee $output_file
+    if [[ "${rc}" != "0" ]]; then
```
Comment (techcobweb) [inline test-scripts/resources-tests.sh:146]: rc is tested but not set ?

File: test-scripts/resources-tests.sh [hunk 4]
```diff
@@ -0,0 +1,551 @@
...
+    cmd="$ORIGINAL_DIR/bin/${binary} properties get --namespace ecosystemtest \
+    --name $prop_name \
+    --bootstrap $bootstrap \
+    --log -"
+
+    info "Command is: $cmd"
+
+    output_file="$ORIGINAL_DIR/temp/resources-update-output.txt"
+    $cmd | tee $output_file
+    if [[ "${rc}" != "0" ]]; then
```
Comment (techcobweb) [inline test-scripts/resources-tests.sh:212]: rc tested but not set.

File: test-scripts/resources-tests.sh [hunk 5]
```diff
@@ -0,0 +1,551 @@
...
+    cmd="$ORIGINAL_DIR/bin/${binary} properties get --namespace ecosystemtest \
+    --infix properties.test \
+    --bootstrap $bootstrap \
+    --log -"
+
+    info "Command is: $cmd"
+
+    output_file="$ORIGINAL_DIR/temp/resources-apply-output.txt"
+    $cmd | tee $output_file
+    if [[ "${rc}" != "0" ]]; then
```
Comment (techcobweb) [inline test-scripts/resources-tests.sh:289]: same . rc not set.

File: test-scripts/resources-tests.sh [hunk 6]
```diff
@@ -0,0 +1,551 @@
...
+    
+    prop_name_to_create="properties.test.create"
+    prop_value_to_create="value-created"
+    #this property needs to already exist
+    prop_name_to_update="properties.test.1"
+    prop_value_to_update="updated-value-2"
+
+    input_file="$ORIGINAL_DIR/temp/resources-apply-input.yaml"
+
+    echo """apiVersion: galasa-dev/v1alpha1
```
Comment (techcobweb) [inline test-scripts/resources-tests.sh:324]: better to use this perhaps ?
```
cat << EOF > $input_file
x
y
z
EOF
```

File: test-scripts/resources-tests.sh [hunk 7]
```diff
@@ -0,0 +1,551 @@
...
+    --bootstrap $bootstrap \
+    -f $input_file \
+    --log -"
+
+    info "Command is: $cmd"
+
+    $cmd
+    rc=$?
+
+    rm $input_file
```
Comment (techcobweb) [inline test-scripts/resources-tests.sh:452]: personally I like to keep the tests against rc as close to the place where rc is set. The rm of the file could happen after everything ?
I guess you are thinking that if it fails I still want everything cleaned up ? But I think if it fails it's good to have the debris around so you can see the input in more detail. Same pattern throughout this file.

**Review by techcobweb — APPROVED** [review]
Some comments about the test script, but looks fine once that is changed.

File: test-scripts/resources-tests.sh (hunk 1 above)
Comment (Akyiaa) [reply test-scripts/resources-tests.sh:295]: This is to check that both properties in the file have been applied (one created, and another updated) and have the expected values

File: test-scripts/resources-tests.sh (hunk 7 above)
Comment (Akyiaa) [reply test-scripts/resources-tests.sh:452]: Yes, I thought it would be better to clean up as soon as possible, but I will change it to be cleaned at it

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**